packages/promotion/src/services/promotion-module.ts

```diff
@@ -360,7 +367,14 @@ export default class PromotionModuleService<
         promotionCodeRulesDataMap.set(promotionData.code, rulesData)
       }
 
-      promotionsData.push(promotionData)
+      if (campaignData) {
+        promotionCodeCampaignMap.set(promotionData.code, campaignData)
+      }
+
+      promotionsData.push({
+        ...promotionData,
+        campaign: campaignId,
+      })
```

Comment:
question: You can only pass one or the other of `campaign_id` and `campaign` no? 

Passing `campaign` should create a new campaign and associate the promotion with it.

Passing `campaign_id` will associate an existing campaign with the promotion.